### PR TITLE
Week 4 Arrays solutions

### PR DESCRIPTION
**Pull request by IreneNakiyingi**
Solutions to Leetcode array week 4 questions

### CODE REVIEW

File: dsa-pros/leetcode/array/JavaScript/ireneNakiyingi/irene_week4.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    return false;
+};
+
+//2.Binary Search
+// time is big O n- Iterate through once. so depends onm length of array
+//space? 
+//run time  52ms ; spce 45.3mb
+//can i use contains/ is in??
+//other option indexOf
+var search = function(nums, target) {
```
Comment (Marienoir) [inline dsa-pros/leetcode/array/JavaScript/ireneNakiyingi/irene_week4.js]: The question requires that `You must write an algorithm with O(log n) runtime complexity.` This solution does not satisfy that requirement even though it works. You might want to check out how the binary search algorithm works. I will approve anyways

File: dsa-pros/leetcode/array/JavaScript/ireneNakiyingi/irene_week4.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
+//Arrays
+//manifesting answswering all questions this time
```
Comment (Marienoir) [inline dsa-pros/leetcode/array/JavaScript/ireneNakiyingi/irene_week4.js:2]: On a lighter note, I like your manifestation here 😀

**Review by Marienoir — APPROVED** [review]

File: dsa-pros/leetcode/array/JavaScript/ireneNakiyingi/irene_week4.js (hunk 1 above)
Comment (IreneNakiyingi) [reply dsa-pros/leetcode/array/JavaScript/ireneNakiyingi/irene_week4.js]: Thanks. I'll look into it as soon as possible.

Comment (IreneNakiyingi) [reply dsa-pros/leetcode/array/JavaScript/ireneNakiyingi/irene_week4.js]: Changes made. I understand now. Thank you.

**Pull request merged**